org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/PreferenceManager.java

```diff
@@ -227,6 +227,9 @@ public void update(Preferences preferences) {
 		Hashtable<String, String> options = JavaCore.getOptions();
 		preferences.updateTabSizeInsertSpaces(options);
 		JavaCore.setOptions(options);
+		List<String> resourceFilters = preferences.getResourceFilters();
+		IEclipsePreferences eclipsePreferences = InstanceScope.INSTANCE.getNode(IConstants.PLUGIN_ID);
+		eclipsePreferences.put(Preferences.JAVA_RESOURCE_FILTERS, String.join("::", resourceFilters));
```

Comment:
Is this here only to pass the data to the `jdt.ls.filesystem` without directly depending on it ? I would just add a comment if that's the case.

Reply:
> Is this here only to pass the data to the jdt.ls.filesystem without directly depending on it ?

Yes, it is.

>  I would just add a comment if that's the case.

See https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2999/commits/f1728dd8416646ccb0104d5c0cd54e7e6028f42a#diff-6fc308461e1c80210cc7e59d8286e3b813e4221804e41cce66fdaaa04f792010R23.